x/superfluid/keeper/twap_price.go

```diff
@@ -37,6 +37,10 @@ func (k Keeper) DeleteEpochOsmoEquivalentTWAP(ctx sdk.Context, denom string) {
 	prefixStore.Delete([]byte(denom))
 }
 
+// Not clear to me how this works?
+// looks like it expects twap to be already in state?
+// but we're supposed to be using spot price instead in MVP?
+// also wouldn't this depend on the ordering of the assets? Osmo/X vs X/Osmo?
 func (k Keeper) GetEpochOsmoEquivalentTWAP(ctx sdk.Context, denom string) sdk.Dec {
```

Comment:
Its set in hooks.go, which contains the epoch code.

Reply:
The function used to calculate it is above in this file, at calculateOsmoBackingPerShare

Reply:
:+1: I see, so by the time someone can send a delegation tx, the twap (spot) value already exists in state. And if it doesn't exist, that indicates that either:
The pool isn't approved for superfluid, the pool doesn't have osmo in it, or the asset isn't an LP token.